lib/puppet/face/config.rb

```diff
@@ -129,4 +129,49 @@
       nil
     end
   end
+
+  action(:delete) do
+    summary _("Delete a Puppet setting.")
+    arguments _("(<setting>")
+    #TRANSLATORS 'main' is a specific section name and should not be translated
+    description "Deletes a setting from the specified section. (The default is the section 'main')."
+    notes <<-'EOT'
+      By default, this action deletes the configuration setting from the 'main'
+      configuration domain. Use the '--section' flags to delete settings from other
+      configuration domains.
+    EOT
+    examples <<-'EOT'
+      Delete the setting 'setting_name' from the 'main' configuration domain:
+
+      $ puppet config delete setting_name
+
+      Delete the setting 'setting_name' from the 'master' configuration domain:
+
+      $ puppet config delete setting_name --section master
+    EOT
+
+    when_invoked do |name, options|
+      options[:section] = options[:section].to_s # If value was left as default - set to default string
```

Comment:
Added this conversion of the section from a symbol to a string, to work with the change in (PUP-2868) which is also up for review

Reply:
Left https://github.com/puppetlabs/puppet/pull/6603#issuecomment-363245919 about updating how section names are handled, since it seems like we shouldn't need to worry about doing this everywhere.